File: pilha/Pilha Estática/Java/Pilha Com Filas/Enfileiravel.java
public interface Enfileiravel {
	void enfileirar(Object elemento);
	Object desenfileirar();
	Object frente();
	Object atualizarInicio();
	Object atualizarFim();	
	
	boolean estaCheia();
	boolean estaVazia();
	String imprimir();
}

File: fila/estatica/Kotlin/Fila Circular/Enfileiravel.kt
interface Enfileiravel {
    // Metodos principais
    fun enfileirar(dado: Any?)  //C
    fun atualizar(dado: Any?)   //U
    fun frente(): Any?          //R
    fun desenfileirar(): Any?   //D    
    // Metodos auxiliares
    fun estaCheia(): Boolean
    fun estaVazia(): Boolean
    fun imprimir(): String
}